src/components/ToastNotification/ToastNotification.module.scss

```diff
@@ -0,0 +1,227 @@
+@import '../../styles/theme';
+
+/* reset */
+button {
+  all: unset;
+}
+
+.ToastRoot {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  border-radius: 0.5rem;
+  padding: 1.5rem 0.5rem 1.5rem 1.5rem;
+  gap: 1.5rem;
+
+  .TextWrapper {
+    display: flex;
+    flex-direction: column;
+    max-width: 14.5rem;
+
+    .ToastTitle {
+      font-weight: 700;
+      font-size: 1rem;
+      line-height: 1.5rem;
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
+
+    .ToastDescription {
+      font-size: 0.75rem;
+      line-height: 0.908rem;
+      color: $color-greyscale-12;
+    }
+  }
+
+  .ToastClose {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 2.25rem;
+    width: 2.25rem;
+
+    &:hover {
+      cursor: pointer;
+    }
+  }
+
+  // primary variants
+  &[data-variant='primary'] {
```

Comment:
What I like to do for variants is set CSS variables, then you apply the CSS variables once. So like

```scss
&[data-variant="primary"] {
    --background: ...;
    --border-color: ...;
    --icon-color: ...;
    --toast-title: ...;
    --toast-close-border-color: ...;
}

&[data-variant="secondary"] {
    --background: ...;
    --border-color: ...;
    --icon-color: ...;
    --toast-title: ...;
    --toast-close-border-color: ...;
}

background: var(--background);
border: 1px solid var(--border-color);

.ToastIcon {
  color: var(--icon-color);
}

.ToastTitle {
  color: var(--toast-title);
}

.ToastClose {
  border-left: 1px solid var(--toast-close-border-color);
}
```

Just a nit though - not necessary for approval.